rasa/utils/tensorflow/rasa_layers.py

```diff
@@ -0,0 +1,815 @@
+                `(batch_size, max_seq_length, input_dim)` where `input_dim` can be
+                different for sparse vs dense tensors.
+            sentence_features: List of 3-D dense or sparse tensors, each with shape
+                `(batch_size, 1, input_dim)` where `input_dim` can be different for
+                sparse vs dense tensors, and can differ from that in `sequence_features`.
+            sequence_feature_lengths: Dense tensor of shape `(batch_size, )` containing
+                the real sequence lengths for the sequence-level features, i.e. the 
+                numbers of real (not padding) tokens for each example in the batch.
+
+    Output shape:
+        outputs: `(batch_size, seq_length, units)` where `units` matches the underlying
+            transformer's output size if the transformer has some layers, otherwise 
+            `units` matches that of the `Ffnn` block applied to the combined features, or
+            it's the output size of the underlying `RasaFeatureCombiningLayer` if the 
+            `Ffnn` block has 0 layers. `seq_length` is the sum of the sequence dimension
+            sizes of sequence- and sentence-level features (for details, see the output
+            shape of `RasaFeatureCombiningLayer`). If both feature types are present,
+            then `seq_length` will be 1 + the length of the longest sequence of real
+            tokens across all examples in the given batch.
+        seq_sent_features: `(batch_size, seq_length, hidden_dim)`, where `hidden_dim` is
+            the output size of the underlying `Ffnn` block, or the output size of the
+            underlying `RasaFeatureCombiningLayer` if the `Ffnn` block has 0 layers.
+        mask_combined_sequence_sentence: `(batch_size, seq_length, hidden_dim)`
+        token_ids: `(batch_size, seq_length, id_dim)` where id_dim is unimportant and 
+            it's the last-dimension size of the first sequence-level dense feature
+            if any is present, and 2 otherwise. Empty tensor if not doing MLM.
+        mlm_boolean_mask: `(batch_size, seq_length, 1)`, empty tensor if not doing MLM.
+        attention_weights: `(transformer_layers, batch_size, num_transformer_heads, 
+            seq_length, seq_length)`, empty tensor if the transformer has 0 layers.
+
+    Raises:
+        A ValueError if no feature signatures for sequence-level features are provided.
+    """
+
+    def __init__(
+        self,
+        attribute: Text,
+        attribute_signature: Dict[Text, List[FeatureSignature]],
+        config: Dict[Text, Any],
+    ) -> None:
+        if not attribute_signature or not attribute_signature.get(SEQUENCE, []):
+            raise ValueError(
+                "The attribute signature must contain some sequence-level feature\
+                signatures but none were found."
+            )
+
+        super().__init__(name=f"rasa_sequence_layer_{attribute}")
+
+        self._tf_layers = {
```

Comment:
could you please explicitly set the type of this dict, so that pycharm stop complaining